### PR TITLE
Add icon to notifications

### LINKED ISSUE #102: Add icon to desktop notification as well as to the in browser notification

### PR DESCRIPTION
**Pull request by DeepDiver1975**
bildschirmfoto von 2017-08-02 17-34-01

fixes #102 & refs #101

NOTE: css magican welcome

### CODE REVIEW

**Comment by PVince81** [comment]
@felixheidecke CSS magician ? ^

File: appinfo/Migrations/Version20170801085340.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+class Version20170801085340 implements ISchemaMigration {
+
+	public function changeSchema(Schema $schema, array $options) {
+		$prefix = $options['tablePrefix'];
+		if ($schema->hasTable("{$prefix}notifications")) {
+			return;
+		}
+
+		// not that valid ....
+		$schemaReader = new MDB2SchemaReader(\OC::$server->getConfig(), \OC::$server->getDatabaseConnection()->getDatabasePlatform());
```
Comment (PVince81) [inline appinfo/Migrations/Version20170801085340.php]: that's cheating 😱

Comment (PVince81) [inline appinfo/Migrations/Version20170801085340.php]: why is this needed just for the icon ? separate PR for proper migration and kill XML file ?

Comment (DeepDiver1975) [inline appinfo/Migrations/Version20170801085340.php]: the icon needs to be stored in the database

**Comment by DeepDiver1975** [comment]
> that's cheating 😱

I know - lazy me ....

**Comment by PVince81** [comment]
schedule more time to do this right ?

We should specify the complete notifications table using migrations, and then extend it with the migration you just added.

File: lib/Notifier.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+use OCP\Notification\INotifier;
+
+class Notifier implements INotifier {
+
+	/**
+	 * @param INotification $notification
+	 * @param string $languageCode The code of the language that should be used to prepare the notification
+	 * @return INotification
+	 * @throws \InvalidArgumentException When the notification was not prepared by a notifier
+	 * @since 9.0.0
```
Comment (PVince81) [inline lib/Notifier.php]: huh, 9.0.0 ? Notifier ?

**Comment by PVince81** [comment]
@VicDeo you'll need to wait for this one before bigintifying the notifications table

**Comment by VicDeo** [comment]
@PVince81 it's not in phase 1: the field that might contain fileid in this app is of type varchar(64)

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 should min-version be 10.0.3 ? didn't we add new icon methods on the interface there ?

I did some defensive coding - it will work with 10.0.2 as well

**Comment by PVince81** [comment]
> defensive coding

kudos for that

**Comment by DeepDiver1975** [comment]
I will backport this - stable10 is still in the release tar ball ....
